site/docs/remote-caching.md

```diff
@@ -0,0 +1,246 @@
+uploads the output files to the CAS, and stores the action result metadata keyed
+by the hash of the action in the action cache.
+
+## HTTP Caching Protocol
+
+Bazel supports remote caching via HTTP/1.1. The protocol is conceivably simple:
+Binary data (BLOB) is uploaded via PUT requests and downloaded via GET requests.
+Action result metadata is stored under the path `/ac/` and output files are
+stored under the path `/cas/`.
+
+For example, consider a remote cache running under
+`http://localhost:8080/cache`. A Bazel request to download action result
+metadata for an action with the SHA256 hash `01ba4719...` will look as follows:
+
+```http
+GET /cache/ac/01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b HTTP/1.1
+Host: localhost:8080
+Accept: */*
+Connection: Keep-Alive
+```
+
+A Bazel request to upload an output file with the SHA256 hash `15e2b0d3…` to the
+CAS will look as follows:
+
+```http
+PUT /cas/15e2b0d3c33891ebb0f1ef609ec419420c20e320ce94c65fbc8c3312448eb225 HTTP/1.1
+Host: localhost:8080
+Accept: */*
+Content-Length: 9
+Connection: Keep-Alive
+
+0x310x320x330x340x350x360x370x380x39
+```
+
+## Configuring Bazel
+
+In order to enable remote caching in Bazel you'll need to specify some flags. We
+recommend adding them to your `~/.bazelrc` file for ease of use.
+
+```
+build --spawn_strategy=remote --genrule_strategy=remote
+build --strategy=Javac=remote --strategy=Closure=remote
+build --remote_http_cache=http://replace-with-your.host:port
+```
+
+The above will enable remote caching but with sandboxing disabled. The support
+for sandboxing with remote caching is currently (as of 0.9.0) experimental but
+works well in our experience.
+
+```
+build --experimental_remote_spawn_cache
+build --remote_http_cache=http://replace-with-your.host:port
+```
+
+## HTTP Caching Backends
+
+A HTTP/1.1 server can treat the data as opaque bytes and thus many existing
+servers can be used as a remote caching backend.
+
+### nginx
+
+nginx with its [WebDAV module] can be used as a remote cache for Bazel. One
+Debian and Ubuntu you can install the `nginx-extras` package. On macOS nginx is
+available via Homebrew:
+
+```bash
+$ brew tap denji/nginx
+$ brew install nginx-full --with-webdav
+```
+
+An example configuration to be put in a `server section` in `nginx.conf` looks
+like below. Note that you will need to change `/path/to/cache/dir` to a valid
+directory where nginx can write to and read from. You may need to change
+`client_max_body_size` option to a larger value in case the cache object is too
+large.
+
+```nginx
+location /cache/ {
+  # The path to the directory where nginx should store the cache contents.
+  root /path/to/cache/dir;
+  # Allow PUT
+  dav_methods PUT;
+  # Allow nginx to create the /ac and /cas subdirectories.
+  create_full_put_path on;
+  # The maximum size of a single file.
+  client_max_body_size 1G;
+  allow all;
+}
+```
+
+### Bazel Remote Cache
```

Comment:
Mention that this is experimental and unsupported.

Reply:
done